src/components/TodoItem.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  isTemp?: boolean;
+  onDelete: (todoId: number) => void;
+  isDeleting?: boolean;
+  onToggleStatus: (todoId: number, newStatus: boolean) => void;
+  isLoading?: boolean;
+  loading?: boolean;
+  editingTodoId: number | null;
+  setEditingTodoId: (id: number | null) => void;
+  onRename: (todoId: number, newTitle: string) => void;
+  setError: (message: string) => void;
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo: { completed, id, title },
+  isTemp = false,
+  onDelete,
+  isDeleting,
+  onToggleStatus,
+  isLoading,
+  loading,
+  editingTodoId,
+  setEditingTodoId,
+  onRename,
+  setError,
+  setLoading,
+}) => {
+  const isEditing = editingTodoId === id;
```

Comment:
Don't pass it as a prop
You need it only here - create a local state `isEditing` here and delete `editingTodoId` in the App.tsx